data_model/model.py

```diff
@@ -47,22 +48,22 @@ def __init__(self):
         self.houses = {}
         self.house_groups = {}
 
-    def add_house(self, house_id):
-        self.houses[house_id] = house_id
+    def add_house(self, house):
+        self.houses[house.id] = house
+        self.houses.update({'id': house})
 
     def add_house_group(self, house_group):
         self.house_groups[house_group.id] = house_group
+        self.house_groups.update({'id': house_group})
 
-    def add_house_to_group(self, house, house_group):
-        # If this method is called multiple times for a specific group, would the first line reset the group to []?
-        self.house_groups[house_group.id] = []
-        self.house_groups[house_group.id] += self.house_groups[house.id]
+    def add_house_to_group(self, house_group):
+        self.house_groups[house_group.id] += self.house_groups
```

Comment:
You currently add a house group to a house group, while you should probably append a house id to the 'houses' attribute of a house group. Suggestion: use an additional class HouseGroup to support e.g. a label/name/description for the group.